src/gui/folderstatusmodel.cpp

```diff
@@ -895,38 +895,36 @@ void FolderStatusModel::slotSetProgress(const ProgressInfo &progress, Folder *f)
 
     auto *pi = &folder._progress;
 
-    const QVector<int> roles = { FolderStatusDelegate::SyncProgressItemString, FolderStatusDelegate::WarningCount, Qt::ToolTipRole };
-
-    switch (progress.status()) {
-    case ProgressInfo::None:
-        Q_UNREACHABLE();
-    case ProgressInfo::Discovery:
-        if (!progress._currentDiscoveredRemoteFolder.isEmpty()) {
-            pi->_overallSyncString = tr("Checking for changes in remote '%1'").arg(progress._currentDiscoveredRemoteFolder);
-            emit dataChanged(index(folderIndex), index(folderIndex), roles);
-        } else if (!progress._currentDiscoveredLocalFolder.isEmpty()) {
-            pi->_overallSyncString = tr("Checking for changes in local '%1'").arg(progress._currentDiscoveredLocalFolder);
-            emit dataChanged(index(folderIndex), index(folderIndex), roles);
-        }
-        break;
-    case ProgressInfo::Reconcile:
-        pi->_overallSyncString = tr("Reconciling changes");
-        emit dataChanged(index(folderIndex), index(folderIndex), roles);
-        break;
-    case ProgressInfo::Propagation:
-        Q_FALLTHROUGH();
-    case ProgressInfo::Done:
-        if (!progress._lastCompletedItem.isEmpty()
-            && Progress::isWarningKind(progress._lastCompletedItem._status)) {
-            pi->_warningCount++;
-        }
+    if (progress.status() == ProgressInfo::Done && !progress._lastCompletedItem.isEmpty()
+        && Progress::isWarningKind(progress._lastCompletedItem._status)) {
+        pi->_warningCount++;
+    }
 
-        // progress updates are expensive, throtle them
-        if (std::chrono::steady_clock::now() - folder._lastProgressUpdated > progressUpdateTimeOutC) {
+    if (std::chrono::steady_clock::now() - folder._lastProgressUpdated > progressUpdateTimeOutC) {
```

Comment:
You lost the comment above this line. I think you should explain your thinking here, though. It'll help future developers greatly to understand why this is implemented the way it is.